contracts/token-distribution/SaleRounds.sol

```diff
@@ -204,115 +159,186 @@ contract SaleRounds is TokenDistribution, GameOwner, ERC20 {
-        //By calculating it this way instead of directly, we don't pay out any incompete periods. instead of: secondsVested * vestingForUserPerSecond
-        ( , uint releasePerFullPeriod) = claimInfo.vestingForUserPerSecond.tryMul(claimInfo.periodGranularity);
-        return periodsVested * releasePerFullPeriod; //this is like "4.55%"
+    function getTotalRemainingForSpecificRound(string calldata _roundType) external view returns(uint256) {
+        RoundType roundType = getRoundTypeByKey(_roundType);
+        return roundDistribution[roundType].totalRemaining;
     }
 
-    function getBalanceToRelease(uint maximalRelease, ClaimInfo memory claimInfo) private pure returns(uint256, uint256) {
-        //technically the precondition / postcondions of the contract prevent this overflow - maybe investigate?
-        require(claimInfo.claimedBalance < claimInfo.balance, "already claimed everything");
-        (,uint unClaimedBalance) = claimInfo.balance.trySub(claimInfo.claimedBalance);
-        require(unClaimedBalance >= 0 , "unsopported unclamined balance");
-        return (unClaimedBalance.min(maximalRelease), unClaimedBalance);
+    function getTotalPending(string calldata _roundType, address _to) external view returns(uint256) {
+        RoundType roundType = getRoundTypeByKey(_roundType);
+
+        return reservedBalances[roundType][_to];
     }
 
-    function getTotalClaimedForAllRounds() public view returns(uint256) {
-       return totalSupply();
+    function getCliffTime(string calldata _roundType) external view onlyGameOwner returns(uint256) {
+        RoundType roundType =  getRoundTypeByKey(_roundType);
+
+        return roundDistribution[roundType].cliff;
     }
 
-    function getTotalRemainingForAllRounds() public view returns(uint256) {
-       (, uint val) = maxSupply.trySub(totalSupply());
-       return val;
+    function mintTokensForExchanges(address _to, uint _amount) public onlyOwner {
+        RoundType roundType = RoundType.EXCHANGES;
+        require(reservedBalances[roundType][_to] >= _amount, "amount is grater then total reserved balance");
+
+        ClaimInfo memory exchangesClaimInfo = ClaimInfo({
+        cliff: roundDistribution[roundType].cliff,
+        vesting: roundDistribution[roundType].vesting,
+        balance: reservedBalances[roundType][_to],
+        claimedBalance: claimedBalances[roundType][_to],
+        periodGranularity: roundDistribution[roundType].vestingGranularity,
+        startTime: roundDistribution[roundType].startTime,
+        secondsVested: 0,
+        vestingForUserPerSecond: 0
+        });
+
+        require(exchangesClaimInfo.balance > 0, "don't have a reserved balance");
+        exchangesClaimInfo.secondsVested = calculateCliffTimeDiff(exchangesClaimInfo);
+        require(exchangesClaimInfo.secondsVested > 0, string.concat("Exchanges round cliff time didn't expired"));
+        exchangesClaimInfo.vestingForUserPerSecond = calculateVestingForUserPerSecond(exchangesClaimInfo);
```

Comment:
This is a Bug. Exchanges allocations do not have ANY vesting and also no claiming!
Just mint the set number of tokens to the wallet (unless there was something that requires us to wait with this minting? seems like a misunderstood requirement - I see the 40/60 split below but I am confused.)

Reply:
https://themoonholding.sharepoint.com/:x:/r/sites/MoonGaming-Leadership/_layouts/15/doc2.aspx?sourcedoc=%7B6cf160b8-2ce8-4a90-ae87-112427305155%7D&action=view&wdinitialsession=40254614-c1b2-42af-b6dd-0f461bd2b004&wdrldsc=5&wdrldc=2&wdrldr=BootTimeMismatch&cid=fb48221d-8f0f-4119-bace-6dd3aa7fb554